layer1/executor/tasks/dkg/mpk_submission.go

```diff
@@ -125,7 +126,22 @@ func (t *MPKSubmissionTask) Execute(ctx context.Context) (*types.Transaction, *t
 
 	// submit if I'm a leader for this task
 	client := t.GetClient()
-	if !utils.AmILeading(client, ctx, logger, int(t.GetStart()), t.StartBlockHash.Bytes(), dkgState.NumberOfValidators, dkgState.Index) {
+	isLeading, err := utils.AmILeading(
+		client,
+		ctx,
+		logger,
+		int(t.GetStart()),
+		t.StartBlockHash.Bytes(),
+		dkgState.NumberOfValidators,
+		dkgState.Index-1,
```

Comment:
Why the -1 here?

Reply:
The ethdkg process is indexed starting at 1, while the vectors that we store on golang are indexed starting at 0. So, I decided to use the index at 0 in the leader election algorithm. I will be adding some comments about this in the code.